hypershift-operator/controllers/nodepool/haproxy.go

```diff
@@ -97,7 +99,20 @@ func (r *NodePoolReconciler) reconcileHAProxyIgnitionConfig(ctx context.Context,
 		return "", true, fmt.Errorf("release image doesn't have a %s image", haProxyRouterImageName)
 	}
 
-	apiServerInternalAddress := config.DefaultAdvertiseAddress
+	// This give support for HTTP Proxy on IPv6 scenarios
+	defaultAddresses := &util.DefaultAdvIps{
+		IPv4: config.DefaultAdvertiseIPv4Address,
+		IPv6: config.DefaultAdvertiseIPv6Address,
+	}
+
+	firstSvcAddresses := util.GetIpsForSvcNetsWithDefaults(hcluster.Spec.Networking.ServiceNetwork, defaultAddresses)
```

Comment:
why is this using the SVC address as the apiServerInternalAddress?

Reply:
If I'm not wrong `config.DefaultAdvertiseAddress` is the first usable address of the first found service network. In the [current implementation](https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/nodepool/haproxy.go#L100) we are setting that as the `apiServerInternalAddress`, so I've modified the code to follow that path, having in mind the changes done.